File: app/src/main/java/com/vincent/commonwidget/helper/FragmentReplaceHelper.java
package com.vincent.commonwidget.helper;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

/**
 * Created by vincent_tung on 2018/2/27.
 */

public class FragmentReplaceHelper {

    private static int mRootLayoutId = 0;

    private static class FragmentReplaceHolder{
        private static FragmentReplaceHelper INSTANCE = new FragmentReplaceHelper();
    }

    public static FragmentReplaceHelper get(){
        return FragmentReplaceHolder.INSTANCE;
    }

    public void setRootLayoutId(int rootLayoutId){
        mRootLayoutId = rootLayoutId;
    }

    public int getRootLayoutId(){
        return mRootLayoutId;
    }

    /**
     * 添加并显示fragment
     * @param fragmentManager
     * @param fragment
     */
    public void addFragment(FragmentManager fragmentManager, Fragment fragment){
        if (mRootLayoutId<=0) throw new RuntimeException("请先设置显示fragment的rootLayoutId！");
        FragmentUtils.addFragment(fragmentManager,fragment, mRootLayoutId,false,true);
    }

    /**
     *
     * @param fm
     */
    public void popFragments(FragmentManager fm){
        FragmentUtils.popFragments(fm);
    }
}

File: app/src/main/java/com/vincent/commonwidget/helper/FragmentReplaceHelper.kt
package com.vincent.commonwidget.helper

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager

/**
 * Created by vincent_tung on 2018/2/23.
 */
class FragmentReplaceHelper {
    private var currentFragment = Fragment()

    private val mManager: FragmentManager? = null

    /**
     * 展示Fragment
     */
    private fun showFragment(fragment: Fragment) {
//        if (currentFragment !== fragment) {
//            val transaction = mManager.beginTransaction()
//            transaction.hide(currentFragment)
//            currentFragment = fragment
//            if (!fragment.isAdded()) {
//                transaction.add(R.id.layout_main_fragment, fragment).show(fragment).commit()
//            } else {
//                transaction.show(fragment).commit()
//            }
//        }
    }
}